app/services/guest_user_service.rb

```diff
@@ -0,0 +1,17 @@
+class GuestUserService < ApplicationService
+  def initialize(cart, params)
```

Comment:
```suggestion
  def initialize(cart, params = {})
```

Assim você pode usar diretamente:
`User.create(@params.merge(guest: true))`